static/js/ui.js

```diff
@@ -122,28 +122,38 @@ function message_hover(message_row) {
     current_message_hover = message_row;
 }
 
-exports.report_message = function (response, status_box, cls) {
+exports.report_message = function (response, status_box, cls, type) {
     if (cls === undefined) {
         cls = 'alert';
     }
 
-    status_box.removeClass(status_classes).addClass(cls)
+    if (type === undefined) {
+        type = ' ';
+    }
+
+    if (type === 'subscriptions-status') {
+        status_box.removeClass(status_classes).addClass(cls).children('#response')
+              .text(response).stop(true).fadeTo(0, 1);
+    } else {
+        status_box.removeClass(status_classes).addClass(cls)
               .text(response).stop(true).fadeTo(0, 1);
+    }
+
```

Comment:
Can you explain why you needed to add an additional argument here?  I'm not sure I see why this couldn't have used the existing framework...